bin/configfile_generator.py

```diff
@@ -0,0 +1,123 @@
+import pprint
+import json
+
+def main():
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "--compact", help="DD4hep compact description xml", required=True
+    )
+    parser.add_argument(
+        "--max_depth", help="Maximum traversal depth of the detector tree", default=10, type=int,
+    )
+    parser.add_argument(
+        "--config_path", help="Location of produced config file", default='nothing', type=str,
+    )
+    parser.add_argument(
+        "--hide_list", help="List of detector geometries that aren't shown", default='', type=str, nargs='+'
+    )
+    parser.add_argument(
+        "--coloring", help="Default colours for detector (choose ild or leave blank)", default='', type=str
+    )
+    args = parser.parse_args()
+
+    ## Gets the detector geometry
+    theDetector = Detector.getInstance()
+    theDetector.fromXML(args.compact)
+    start = theDetector.world()
+
+    ## Runs through the detector to make a tree dictionary of the detector parts
+    detector_dict = tree(start, 0, args.max_depth)
+    ## Processes the detector tree to make a usable config file
+    subPart_processed, hidden_children = post_processing(detector_dict, list(detector_dict.keys()), '|'.join(args.hide_list[0].split(", ")), args.coloring)
+    ## Produce a config.json file using the edited detector tree
+    produce_config(subPart_processed, hidden_children, args.config_path)
+
+def process_name(raw_name):
+    ## Changes any numbers for parts of a detector into .* (which in regex can mean any ending)
+    name = re.sub(r"\d+", ".*", raw_name)
+    return name
+
+def add_ild_colors(subdetector):
+    subdetector_dict = {"Tube": [1,0.7,0.5],
+                        "BeamPipe": [0,1,0],
+                        "Vac": [0,0,0],
+                        "VXD": [0.1,0.5,0.5],
+                        "FTD": [0.39,0.1,0.57],
+                        "BeamCal": [1,1,1],
+                        "SIT": [0.86,0.86,0.86],
+                        "SET": [0.86, 0.86, 0.86],
+                        "TPC": [0.96,0.95,0],
+                        "Ecal": [0.48,0.95,0],
+                        "Hcal": [0.76,0.76,0.19],
+                        "Yoke": [0.09,0.76,0.76],
+                        "Coil": [0.28,0.28,0.86],
+                        "Fcal": [0.67, 0.66, 0.67]
+                        }
+    for s in subdetector_dict:
+        find_subdetector = re.search(f'{s}', f'{subdetector}',re.IGNORECASE)
+        if find_subdetector: return subdetector_dict[s]
+    if find_subdetector == None: return [0.57,0.63,0.81]
+
+
+def tree(detElement, depth, maxDepth):
+    ## Creates a detector tree using the geometry while also being able to set a max depth that the config file reaches
+    nd = {}
+    depth += 1
+    children = detElement.children()
+    for raw_name, child in children:
+        if depth > maxDepth: tree(child, depth, maxDepth)
+        else: dictionary = tree(child, depth, maxDepth); nd.update({raw_name: dictionary})
```

Comment:
```suggestion
        if depth > maxDepth: 
            tree(child, depth, maxDepth)
        else: 
            dictionary = tree(child, depth, maxDepth)
            nd.update({raw_name: dictionary})
```
Makes it a bit easier to read.